### PR TITLE
KOGITO-4919: [DMN Designer] SDM is broken on -kogito-webapp-runtime

### PR DESCRIPTION
**Pull request by karreiro**
**JIRA**: [KOGITO-4919](https://issues.redhat.com/browse/KOGITO-4919): [DMN Designer] SDM is broken on -kogito-webapp-runtime

--

**Part of an ensemble:**
- https://github.com/kiegroup/kie-wb-common/pull/3611
- https://github.com/kiegroup/drools-wb/pull/1484
- https://github.com/kiegroup/kie-wb-distributions/pull/1097

--

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* Retest PR: <b>jenkins retest this</b>
* A full downstream build: <b>jenkins do fdb</b>
* A compile downstream build: <b>jenkins do cdb</b>
* A full production downstream build: <b>jenkins do product fdb</b>
* An upstream build: <b>jenkins do upstream</b>
</details>

### CODE REVIEW

**Review by danielzhe — APPROVED** [review]

**Review by romartin — APPROVED** [review]

**Review by romartin — CHANGES_REQUESTED** [review]
Hey @karreiro 

I have this fix locally, and it works for `kie-wb-common`, but now trying to build `drools-wb`, it fails because of:

`[ERROR] Failed to execute goal org.codehaus.mojo:gwt-maven-plugin:2.9.0:compile (gwt-compile) on project drools-wb-webapp: GWT Module org.kie.dmn.feel.gwt.functions.DMNFeelGWTFunctions not found in project sources or resources. -> [Help 1]
`
So maybe some fix on some pom is still necessary... I understand if `drools-wb `doesn't compile, also `kie-wb-distributions` will not compile... any clues?

**Comment by romartin** [comment]
jenkins retest this

File: kie-wb-common-dmn/kie-wb-common-dmn-client/pom.xml [hunk 1]
```diff
@@ -281,6 +282,7 @@
     <dependency>
       <groupId>org.kie</groupId>
       <artifactId>kie-dmn-feel-gwt</artifactId>
+      <scope>provided</scope>
```
Comment (romartin) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/pom.xml:285]: @karreiro also not sure if makes sense this change - when classifier is `sources`, I think doesn't make sense to use the `provided` scope. The sources are only necessary for the gwt compiler, so the artifact is needed at compile scope, so making it runtime makes things not work.

**Review by romartin — COMMENTED** [review]
Another issue found, look at my comment below. By not changing the scope for the sources artifact,  maybe then everything works... will let to you guys to test it, I'm still with the issue on DMN too.

Thanks!

**Comment by karreiro** [comment]
Thank you, @romartin! :-) Changes applied.

**Comment by romartin** [comment]
@karreiro just checked out the branch again and the SDM for webapp-kogito-runtime does't work now :( 
Same issues as initially, libraries are ending up in the web-inf/lib again and jetty doesn't start

**Comment by romartin** [comment]
jenkins do fdb

**Review by romartin — APPROVED** [review]
Hey @karreiro 

Tested both DMN for webapp-kogito-runtime and also drools-wb-webapp (WAR, not SDM), both apps working properly. So looks good to me!! Thanks again for the fixes, great job! :+1:

**Comment by karreiro** [comment]
jenkins do fdb

**Comment by karreiro** [comment]
jenkins do fdb

**Comment by karreiro** [comment]
jenkins do fdb

**Comment by karreiro** [comment]
@romartin @danielzhe I just incorporated I small-new commit to fix a CI error.

**Comment by karreiro** [comment]
jenkins do fdb

**Comment by karreiro** [comment]
jenkins retest this

**Comment by karreiro** [comment]
jenkins do fdb

**Comment by danielzhe** [comment]
I'm not 100% sure, but I think this fix broke the regular DMN Showcase. Can you try to run DMN Showcase with this fix applied? By showcase I mean 'kie-wb-common-dmn-webapp-standalone'. @karreiro

**Comment by karreiro** [comment]
Thank you, @danielzhe! Double-checking it :-)

**Comment by karreiro** [comment]
jenkins do fdb

**Comment by karreiro** [comment]
@danielzhe fixes applied, the branch is rebased, and ready for the re-review :-)

**Review by danielzhe — APPROVED** [review]
It's fixed now! Thank you, @karreiro !

**Comment by karreiro** [comment]
jenkins do fdb

**Comment by karreiro** [comment]
jenkins do fdb

**Comment by karreiro** [comment]
jenkins do fdb

**Comment by karreiro** [comment]
jenkins do fdb

**Pull request merged**